### PR TITLE
Big objects creation tests

### PR DESCRIPTION
**Pull request by geonexus**
### REVIEWERS

@flopezag @jframos 
### DESCRIPTION

Added test to verify that is possible to upload big files to the object-store (more than 5mb)

### CODE REVIEW

File: fiware-region-sanity-tests/commons/constants.py [hunk 1]
```diff
@@ -67,10 +67,15 @@
...
 TEST_CONTAINER_PREFIX = "testing_container"
 TEST_TEXT_OBJECT_PREFIX = "testing_text_object"
+TEST_BIG_OBJECT_PREFIX = "test_big_file"
 TEST_TEXT_FILE_EXTENSION = ".txt"
+TEST_BIG_FILE_EXTENSION = ".zip"
 
 # SWIFT CONTANTS
 SWIFT_RESOURCES_PATH = "resources/swift_objects/"
+BIG_FILE_URL_1 = "https://github.com/telefonicaid/fiware-paas/archive/1.1.0.zip"
+BIG_FILE_URL_2 = "https://github.com/telefonicaid/fiware-paas/archive/1.3.4.zip"
```
Comment (flopezag) [inline fiware-region-sanity-tests/commons/constants.py]: A ZIP file which is a binary and a video is upload to the github? There is no other possibility?

File: fiware-region-sanity-tests/commons/constants.py [hunk 2]
```diff
@@ -88,6 +93,3 @@
 
 # ENDPOINTS
 ENDPOINT_TYPE_PUBLIC_URL = "publicURL"
-
-# TYPES
-TEST_TEXT_CONTENT_TYPE = "text/plain"
```
Comment (flopezag) [inline fiware-region-sanity-tests/commons/constants.py:93]: Why do you delete this line, we do not need the test associated to the upload of a text file in the Object Storage?

File: fiware-region-sanity-tests/commons/swift_operations.py [hunk 3]
```diff
@@ -27,7 +27,7 @@
 from swiftclient import client
 from commons.constants import DEFAULT_REQUEST_TIMEOUT, OBJECT_STORE_MAX_RETRIES, PROPERTIES_CONFIG_CRED_KEYSTONE_URL, \
     PROPERTIES_CONFIG_CRED_USER, PROPERTIES_CONFIG_CRED_PASS, PROPERTIES_CONFIG_CRED_TENANT_ID, SERVICE_SWIFT_NAME, \
-    ENDPOINT_TYPE_PUBLIC_URL, PROPERTIES_CONFIG_CRED_USER_DOMAIN_NAME, TEST_TEXT_CONTENT_TYPE
+    ENDPOINT_TYPE_PUBLIC_URL, PROPERTIES_CONFIG_CRED_USER_DOMAIN_NAME
```
Comment (flopezag) [inline fiware-region-sanity-tests/commons/swift_operations.py:30]: Idem

File: fiware-region-sanity-tests/commons/constants.py (hunk 2 above)
Comment (geonexus) [inline fiware-region-sanity-tests/commons/constants.py:93]: Finally, swift client accepts text_files and binary_files in the same way, so it is unnecessary to specify the content_type.

File: fiware-region-sanity-tests/commons/swift_operations.py (hunk 3 above)
Comment (geonexus) [inline fiware-region-sanity-tests/commons/swift_operations.py:30]: It is no needed anymore

File: fiware-region-sanity-tests/commons/constants.py (hunk 1 above)
Comment (geonexus) [inline fiware-region-sanity-tests/commons/constants.py]: The ZIP file is not uploaded to github. 
This URL is an example binary file to test big files uploading to an Object Store instance. Those files are downloaded from github.

File: fiware-region-sanity-tests/commons/constants.py (hunk 2 above)
Comment (flopezag) [inline fiware-region-sanity-tests/commons/constants.py:93]: NTC

File: fiware-region-sanity-tests/commons/swift_operations.py (hunk 3 above)
Comment (flopezag) [inline fiware-region-sanity-tests/commons/swift_operations.py:30]: NTC

File: fiware-region-sanity-tests/commons/constants.py (hunk 1 above)
Comment (jframos) [inline fiware-region-sanity-tests/commons/constants.py]: That file is for "constants". I think those values are configuration parameters. Perhaps we can put them in the `settings.json` file.

Comment (flopezag) [inline fiware-region-sanity-tests/commons/constants.py]: +1

File: fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py [hunk 4]
```diff
@@ -160,8 +162,78 @@ def test_delete_an_object_from_a_container(self):
...
+        origin2 = hashlib.md5(open(SWIFT_RESOURCES_PATH + TEST_BIG_OBJECT_PREFIX + suffix2 + TEST_BIG_FILE_EXTENSION,
+                                   'rb')
+                              .read()).hexdigest()
+
+        self.assertNotEqual(origin2, remote, "The second file and the downloaded file are the same, "
+                                             "it cannot be possible")
+
+        os.remove(SWIFT_RESOURCES_PATH + big_object_name)
+        os.remove(SWIFT_RESOURCES_PATH + TEST_BIG_OBJECT_PREFIX + TEST_BIG_FILE_EXTENSION)
+        os.remove(SWIFT_RESOURCES_PATH + TEST_BIG_OBJECT_PREFIX + suffix2 + TEST_BIG_FILE_EXTENSION)
```
Comment (jframos) [inline fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py]: Should we put this code in the "cleanUp" procedure? If not, what will happen if some exception is raised between the file creation and this lines?

File: fiware-region-sanity-tests/commons/swift_operations.py [hunk 5]
```diff
@@ -94,21 +94,20 @@ def create_container(self, container_name):
...
-        Creates a new text object into a container
+        Creates a new object into a container
         :param container_name: Name of the container
         :return: None if container was created and the error message if something failed.
         """
         with open(local_file, 'r') as test_file:
             response = self.client.put_object(container_name, object_name,
-                                        contents=test_file.read(),
-                                        content_type=TEST_TEXT_CONTENT_TYPE)
+                                        contents=test_file.read())
```
Comment (jframos) [inline fiware-region-sanity-tests/commons/swift_operations.py]: pep8-compliant?

File: fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py [hunk 6]
```diff
@@ -91,7 +92,7 @@ def test_create_text_object_and_download_it(self):
 
         ## Uploading the object
         try:
-            response = self.swift_operations.create_text_object(containerName, SWIFT_RESOURCES_PATH
+            response = self.swift_operations.create_object(containerName, SWIFT_RESOURCES_PATH
                                                                 + TEST_TEXT_OBJECT_PREFIX + TEST_TEXT_FILE_EXTENSION,
```
Comment (jframos) [inline fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py]: pep8-compliant?

File: fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py [hunk 7]
```diff
@@ -139,7 +142,7 @@ def test_delete_an_object_from_a_container(self):
...
```
Comment (jframos) [inline fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py]: pep8-compliant?

File: fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py [hunk 8]
```diff
@@ -160,8 +162,78 @@ def test_delete_an_object_from_a_container(self):
             self.fail(ex)
 
         try:
-            self.swift_operations.get_text_object(containerName, textObjectName,
+            self.swift_operations.get_object(containerName, textObjectName,
                                                              SWIFT_RESOURCES_PATH)
```
Comment (jframos) [inline fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py:151]: pep8-compliant?

Comment (geonexus) [inline fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py:151]: yes

File: fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py (hunk 7 above)
Comment (geonexus) [inline fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py]: yes

File: fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py (hunk 6 above)
Comment (geonexus) [inline fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py]: yes

File: fiware-region-sanity-tests/commons/swift_operations.py (hunk 5 above)
Comment (geonexus) [inline fiware-region-sanity-tests/commons/swift_operations.py]: yes

File: fiware-region-sanity-tests/resources/settings.json [hunk 9]
```diff
@@ -11,6 +11,10 @@
     "test_configuration": {
         "phonehome_endpoint": ""
     },
+    "swift_configuration": {
```
Comment (jframos) [inline fiware-region-sanity-tests/resources/settings.json]: I think it is better to put this config inside the _"test_configuration"_ property.

Comment (geonexus) [inline fiware-region-sanity-tests/resources/settings.json]: fixed

**Comment by jframos** [comment]
LGTM

**Comment by flopezag** [comment]
LGTM

**Pull request merged**